frontend/src/pages/ResetPasswordPage.tsx

```diff
@@ -0,0 +1,160 @@
+import { useState, FormEvent } from 'react';
+import { useDispatch } from 'react-redux';
+import { Trans, useTranslation } from 'react-i18next';
+import { Alert } from 'react-bootstrap';
+import { Link, useHistory, useParams } from 'react-router-dom';
+import classNames from 'classnames';
+import { userActions } from '../redux/user';
+import { ModalContent } from '../components/modals/modalparts/ModalContent';
+import { ModalHeader } from '../components/modals/modalparts/ModalHeader';
+import { Input, errorState } from '../components/forms/FormField';
+import { Footer } from '../components/Footer';
+import { paths } from '../routers/paths';
+import { StoreDispatchType } from '../redux';
+import { ReactComponent as CheckIcon } from '../icons/check_icon.svg';
+import css from './ResetPasswordPage.module.scss';
+
+const classes = classNames.bind(css);
+
+export const ResetPasswordPage = () => {
+  const dispatch = useDispatch<StoreDispatchType>();
+  const history = useHistory();
+  const { t } = useTranslation();
+  const { token } = useParams<{
+    token: string | undefined;
+  }>();
+
+  const [view, setView] = useState('Reset password');
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const confirmationError = {
+    err: errorState(useState('')),
+  };
+
+  const handleReset = (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    e.stopPropagation();
+    if (password !== confirmPassword) {
+      confirmationError.err.setMessage(t('Password confirmation error'));
+      return;
+    }
+
+    if (!token) return;
+
+    dispatch(userActions.resetPassword({ token, password })).then((res) => {
+      if (userActions.resetPassword.rejected.match(res)) {
+        if (res.payload?.response?.status === 403)
+          setErrorMessage(t('Token expired'));
+        else if (res.payload?.response?.status === 404)
+          setErrorMessage(t('Token not found'));
+        else if (res.payload) {
+          setErrorMessage(t('Error message', { error: res.payload.message }));
+        }
+      } else {
+        setView('Done');
+      }
+    });
+  };
+
+  const resetPasswordView = () => {
+    return (
+      <>
+        <div className="formDiv">
```

Comment:
```suggestion
  const resetPasswordView = () => (
      <div className="formDiv">
```

At least the unnecessary fragment should be removed